libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1323,7 +1322,6 @@ void SMTEncoder::indexOrMemberAssignment(Expression const& _expr, smtutil::Expre
 				smtutil::Expression(make_shared<smtutil::SortSort>(smt::smtSort(*baseType)), baseType->toString(true)),
 				{smtutil::Expression::store(symbArray->elements(), indexExpr, toStore), symbArray->length()}
 			);
-			m_context.expression(*indexAccess)->increaseIndex();
```

Comment:
Why remove this?

Reply:
To me it looks redundant. If an expression already exists then the call to `defineExpr` on the next line will increase the index before assigning the value.
If the expression did not exist before this point then there is no point on increasing the index, no?

Reply:
`defineExpr` doesn't call `increaseIndex`, it just assigns the current variable

Reply:
`defineExpr` calls `createExpr` which calls `m_context.createExpression` which just increases the index if it is already a known expression.

Reply:
Ah right